### ISSUE TITLE
Remove (fixed) dependency on psych

### ISSUE THREAD

**Issue by nathansamson** [troessner/reek#1531]
Trying to install a rails project with reek 6.0.0 on windows fails because psych is not compatible with Ruby 2.6 or newer on windows.
As the latest release of psych is almost 18 months old (december 2018), and no comments on the bug report on their end (https://github.com/ruby/psych/issues/405) I propose dropping the dependency on psych. Its also almost not referenced in the codebase, and should be part of MRI anyway, I don't think this should be an issue (it could be a dependeny for jruby still?)

    Bundler found conflicting requirements for the Ruby version:
    In Gemfile:
      Ruby (~> 2.7.1.0) x64-mingw32

    reek (>= 5.6.0) x64-mingw32 was resolved to 6.0.0, which depends on
      psych (~> 3.1.0) x64-mingw32 was resolved to 3.1.0, which depends on
        Ruby (< 2.6) x64-mingw32`

**Comment by mvz** [comment]
Looking at the commit that introduced Psych, it seems we need it as a real dependency to support YAML loading with the new option syntax on Ruby 2.4 and 2.5.

You can try installing the platform-independent version of Psych 3.1.0 first (`gem install psych --platform ruby`), or add an equivalent line to the Gemfile.

**Comment by crcraig92** [comment]
It's possible this was resolved but still causing issues because the dependency is ~> and the change was minor version not a patch.  I tried many things to force the "install" to be recognized but it would not work.  psych 3.1.0 is a default gem that is packaged with Ruby 2.7.1 and is there from a fresh build with MSYS2.  I have forced the platform builds and still received the error.  I have updated to 3.2.0 and still received the issue.  As a default gem, is it still required to be listed as a runtime dependency? Maybe remove the dependency or >= 3.1.0?  Otherwise I would say there is a Ruby version <2.6 for this gem, if nathansamson is correct.  I upgraded from 2.5 to 2.7 so I have not verified compatibility with 2.6.

`Gem::RuntimeRequirementNotMetError: psych requires Ruby version >=2.3, < 2.6.
The current ruby version is 2.7.1.83.
An error occurred while installing psych (3.1.0), and Bundler
cannot continue.
Make sure that `gem install psych -v '3.1.0' --source 
'[redacted]'` succeeds
before bundling.

In Gemfile:
  rubycritic was resolved to 4.1.0, which depends on
     reek was resolved to 5.6.0, which depends on
       psych`

**Comment by mvz** [comment]
>  As a default gem, is it still required to be listed as a runtime dependency?

Yes, because we need the non-default version on Ruby 2.4 and 2.5.

> I tried many things to force the "install" to be recognized but it would not work. psych 3.1.0 is a default gem that is packaged with Ruby 2.7.1 and is there from a fresh build with MSYS2.

If psych 3.1.0 is already installed with Ruby 2.7.1 there shouldn't be any problem!? Can you please post the result of the following commands:

```
ruby --version
gem list psych
gem install reek
```

**Comment by mvz** [comment]
I've just released Reek 6.0.2 which loosens the dependency on Psych. This should resolve this issue but I cannot test it directly. Please try and let me know.

**Comment by crcraig92** [comment]
``` λ ruby --version
ruby 2.7.1p83 (2020-03-31 revision a0c7c23c9c) [x64-mingw32]

λ gem install reek
Fetching reek-6.0.2.gem
Successfully installed reek-6.0.2
Parsing documentation for reek-6.0.2
Installing ri documentation for reek-6.0.2
Done installing documentation for reek after 7 seconds
1 gem installed
```
It seems fixed with the 6.0.2 release.  Sorry for the delay, I cannot verify what the output of those commands were prior to this but psych would ask me to specifically install version '3.1.0' which as you said, is the default and should have been installed and recognized.  It was installed, and I don't understand why it wasn't recognized.  
```λ gem install psych -v '3.1.0'
ERROR:  Error installing psych:
        The last version of psych (= 3.1.0) to support your Ruby & RubyGems was 3.1.0. Try installing it with `gem install psych -v 3.1.0`
        psych requires Ruby version >= 2.3, < 2.6. The current ruby version is 2.7.1.83.
```
So again, sorry for the delay, but yes it does appear to be working.  The original issue was installing rubycritic which now installs on Ruby 2.7.
Thank you.

**Comment by mvz** [comment]
Thanks for checking this, @crcraig92.
